### PR TITLE
Introduce AxisInternal

### PR DESCRIPTION
**Pull request by masayuki0812**
This PR introduces `AxisInternal` and makes it possible to access axis internal functions.

### CODE REVIEW

**Comment by kt3k** [comment]
I agree with the structure of this refactoring.

As we talked informally on chat, I prefer using `this` and arrow functions instead of `$$$` convention. (`$$$` seems a bit strange to me.)

File: src/axis.js [hunk 1]
```diff
@@ -1,16 +1,386 @@
...
+        else {
+            if (!arguments.length) { return $$$.tickValues; }
+            $$$.tickValues = x;
+        }
+        return axis;
+    };
+    return axis;
+};
+
+export default function Axis(owner) {
```
Comment (kt3k) [inline src/axis.js]: How about using class keyword like the below?

```js
export default class Axis extends Component {
  constructor (owner) {
    ...
  }
}
```

By the above we don't need to maintain `inherit` util anymore.

Comment (masayuki0812) [inline src/axis.js]: Looks good. Implemented by f1f8ab60a570ae72fd5af005b14289a91e410594 .

**Comment by masayuki0812** [comment]
Fixed `$$$` and changed to use `class` and `extends` instead of `inherit`.

**Pull request merged**